src/jit/codegenxarch.cpp

```diff
@@ -1329,9 +1345,23 @@ void CodeGen::genCodeForDivMod(GenTreeOp* treeNode)
     }
     else
     {
-        // dividend must be in RAX
-        if (dividend->gtRegNum != REG_RAX)
+        if (dividendIsLong)
+        {
+            assert(dividendLo != nullptr && dividendHi != nullptr);
+
+            // dividendLo must be in RAX; dividendHi must be in RDX
+            if (dividendLo->gtRegNum != REG_RAX)
```

Comment:
I *think* it's just paranoia, but I was copying the logic that was already here.

The div will kill both EAX and EDX. But I don't know about the GC question.